src/routes/session.rs

```diff
@@ -101,3 +106,112 @@ pub fn delete(mut cookies: Cookies) -> Redirect {
     }
     Redirect::to("/")
 }
+
+pub struct ResetRequest {
+    pub mail: String,
+    pub id: String,
+    pub creation_date: Instant,
+}
+
+#[get("/password-reset")]
+pub fn password_reset_request_form(conn: DbConn, intl: I18n) -> Ructe {
+    render!(session::password_reset_request(
+        &(&*conn, &intl.catalog, None),
+        &ResetForm::default(),
+        ValidationErrors::default()
+    ))
+}
+
+#[derive(FromForm, Validate, Default)]
+pub struct ResetForm {
+    #[validate(email)]
+    pub email: String,
+}
+
+#[post("/password-reset", data = "<form>")]
+pub fn password_reset_request(
+    conn: DbConn,
+    intl: I18n,
+    mail: State<Arc<Mutex<Mailer>>>,
+    form: Form<ResetForm>,
+    requests: State<Arc<Mutex<Vec<ResetRequest>>>>
+) -> Ructe {
+    if User::find_by_email(&*conn, &form.email).is_ok() {
+        let id = plume_common::utils::random_hex();
+        {
+            let mut requests = requests.lock().unwrap();
+            requests.push(ResetRequest {
```

Comment:
to prevent the `State<...>` from growing up in memory (basically be a memory leak), you should filter and eliminate old entries either with the internal scheduler, or somewhere like here